internal/databases/postgres/backup_push_handler.go

```diff
@@ -304,33 +304,46 @@ func (bh *BackupHandler) uploadBackup() internal.TarFileSets {
 // HandleBackupPush handles the backup being read from Postgres or filesystem and being pushed to the repository
 // TODO : unit tests
 func (bh *BackupHandler) HandleBackupPush() {
-	folder := bh.Workers.Uploader.UploadingFolder
-	baseBackupFolder := folder.GetSubFolder(bh.Arguments.backupsFolder)
-	tracelog.DebugLogger.Printf("Base backup folder: %s", baseBackupFolder)
-
 	bh.CurBackupInfo.StartTime = utility.TimeNowCrossPlatformUTC()
 
 	if bh.Arguments.pgDataDirectory == "" {
-		if bh.Arguments.forceIncremental {
-			tracelog.ErrorLogger.Println("Delta backup not available for remote backup.")
-			tracelog.ErrorLogger.Fatal("To run delta backup, supply [db_directory].")
-		}
-		// If no arg is parsed, try to run remote backup using pglogrepl's BASE_BACKUP functionality
-		tracelog.InfoLogger.Println("Running remote backup through Postgres connection.")
-		tracelog.InfoLogger.Println("Features like delta backup are disabled, there might be a performance impact.")
-		tracelog.InfoLogger.Println("To run with local backup functionalities, supply [db_directory].")
-		if bh.PgInfo.pgVersion < 110000 && !bh.Arguments.verifyPageChecksums {
-			tracelog.InfoLogger.Println("VerifyPageChecksums=false is only supported for streaming backup since PG11")
-			bh.Arguments.verifyPageChecksums = true
-		}
-		bh.createAndPushRemoteBackup()
-		return
+		bh.handleBackupPushRemote()
+	} else {
+		bh.handleBackupPushLocal()
 	}
+}
 
-	if utility.ResolveSymlink(bh.Arguments.pgDataDirectory) != bh.PgInfo.PgDataDirectory {
-		tracelog.ErrorLogger.Panicf("Data directory read from Postgres (%s) is different than as parsed (%s).",
-			bh.Arguments.pgDataDirectory, bh.PgInfo.PgDataDirectory)
+func (bh *BackupHandler) handleBackupPushRemote() {
+	if bh.Arguments.forceIncremental {
+		tracelog.ErrorLogger.Println("Delta backup not available for remote backup.")
+		tracelog.ErrorLogger.Fatal("To run delta backup, supply [db_directory].")
 	}
+	// If no arg is parsed, try to run remote backup using pglogrepl's BASE_BACKUP functionality
+	tracelog.InfoLogger.Println("Running remote backup through Postgres connection.")
+	tracelog.InfoLogger.Println("Features like delta backup are disabled, there might be a performance impact.")
+	tracelog.InfoLogger.Println("To run with local backup functionalities, supply [db_directory].")
+	if bh.PgInfo.pgVersion < 110000 && !bh.Arguments.verifyPageChecksums {
+		tracelog.InfoLogger.Println("VerifyPageChecksums=false is only supported for streaming backup since PG11")
+		bh.Arguments.verifyPageChecksums = true
+	}
+	bh.createAndPushRemoteBackup()
+}
+
+func (bh *BackupHandler) handleBackupPushLocal() {
+	{
```

Comment:
Why wrap this block in curlies?

Reply:
a habit I have to scope variables that are short lived. For example : we *know* that `fromServer` and `fromCli` have no impact after `}`.

If you think this is weird feel free to remove them.